### PR TITLE
Fix for handling .id=NULL in rdply and ldply

### PR DESCRIPTION
**Pull request by tentacles-from-outer-space**
Following http://stackoverflow.com/q/25733636/168747 I prepare fix for this.

rdply(5, function() data.frame(a=1), .id=NULL)
ldply(1:5, function(i) data.frame(a=1), .id=NULL)

should work. I add few tests to handle corner cases.

### CODE REVIEW

File: R/ldply.r [hunk 1]
```diff
@@ -18,7 +18,7 @@ ldply <- function(.data, .fun = NULL, ..., .progress = "none", .inform = FALSE,
     .progress = .progress, .inform = .inform,
     .parallel = .parallel, .paropts = .paropts)
 
-  if (is.na(.id)) {
+  if (!is.null(.id) && is.na(.id)) {
```
Comment (hadley) [inline R/ldply.r]: This might be better written as `identical(.id, NA)`

File: R/rdply.r [hunk 2]
```diff
@@ -31,8 +30,12 @@ rdply <- function(.n, .expr, .progress = "none", .id = NA) {
   res <- .rlply_worker(.n, .progress,
                        eval.parent(substitute(function() .expr)))
   names(res) <- seq_len(.n)
-  labels <- data.frame(.n = seq_len(.n))
-  if (!is.na(.id))
-    names(labels) <- .id
+  if (is.null(.id))
```
Comment (hadley) [inline R/rdply.r]: Can you please wrap all blocks in `{}` with appropriate spacing. Full style guide at http://r-pkgs.had.co.nz/r.html#style

**Comment by hadley** [comment]
A few minor comments. Could you please merge/rebase with master?

**Comment by dougmitarotonda** [comment]
I am happy to help get wrapped up and pulled! I must admit I have not yet had to rebase a repository so I will have to read into how that works. @tentacles-from-outer-space let me know if you would rather make the changes @hadley suggested.

**Comment by dougmitarotonda** [comment]
See https://github.com/hadley/plyr/pull/235 for my attempt to resolve this and get the code pulled.
